src/jit/codegenxarch.cpp

```diff
@@ -6789,30 +6805,60 @@ void         CodeGen::genJumpKindsForTreeLongLo(GenTreePtr    cmpTree,
-//                      cmp loOp1,loOp2
+//    GT_GT; unsigned   cmp hiOp1,hiOp2         If hiOp1 is not equal to hiOp2, the entire compare
+//                      jne label               is false. Otherwise, we need to compare the lo
+//                      cmp loOp1,loOp2         halves
 //                      label: seta
 //
-//    GT_GE             cmp hiOp1,hiOp2         If hiOp1 is greater than hiOp2, the entire compare
-//                      ja label                is true. If hiOp1 is less than hiOp2, the entire
-//                      jne label               compare is false.
-//                      cmp loOp1,loOp2
+//    GT_GE; unsigned   cmp hiOp1,hiOp2         If hiOp1 is not equal to hiOp2, the entire compare
+//                      jne label               is false. Otherwise, we need to compare the lo
+//                      cmp loOp1,loOp2         halves
 //                      label: setae
 //
+// For signed long comparisons, we need additional labels, as we need to use signed sets:
+//
+//    GT_LT; signed     cmp hiOp1,hiOp2         If hiOp1 is not equal to hiOp2, the entire compare
+//                      jne labelHi             is false. Otherwise, we need to compare the lo
+//                      cmp loOp1,loOp2         halves
+//                      setb                    Unsigned set for lo compare
+//                      jmp labelFinal
+//                      labelHi: setl           Signed set for high compare
+//                      labelFinal:
+//
+//    GT_LE; signed     cmp hiOp1,hiOp2         If hiOp1 is not equal to hiOp2, the entire compare
+//                      jne labelHi             is false. Otherwise, we need to compare the lo
+//                      cmp loOp1,loOp2         halves
+//                      setbe                   Unsigend set for lo compare
+//                      jmp labelFinal
+//                      labelHi: setle          Signed set for hi compare
+//                      labelFinal:
+//
+//    GT_GT; signed     cmp hiOp1,hiOp2         If hiOp1 is not equal to hiOp2, the entire compare
+//                      jne labelHi             is false. Otherwise, we need to compare the lo
+//                      cmp loOp1,loOp2         halves
+//                      seta                    Unsigned set for lo compare
+//                      jmp labelFinal
+//                      labelHi: setg           Signed set for high compare
+//                      labelFinal
+//
+//    GT_GE; signed     cmp hiOp1,hiOp2         If hiOp1 is not equal to hiOp2, the entire compare
+//                      jne labelHi             is false. Otherwise, we need to compare the lo
+//                      cmp loOp1,loOp2         halves
+//                      setae                   Unsigned set for lo compare
+//                      jmp labelFinal
+//                      labelHi: setge          Signed set for hi compare
+//                      labelFinal:
+//
```

Comment:
The function header comment doesn't say anything about handling the branch cases, and the kind of codegen that is done for those.